File: client-scalaz/src/main/scala/org/openapitools/client/api/AlbumContestListResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import AlbumContestListResponse._

case class AlbumContestListResponse (
  valid: Option[Boolean],
message: Option[String],
version: Option[Double],
serializeNulls: Option[Boolean],
startTimeLog: Option[Long],
errorCode: Option[String],
request: Option[List[NameStringValueResponse]],
items: Option[List[AlbumContestResponse]],
hasMoreResults: Option[Boolean],
count: Option[Integer],
countTotal: Option[Long],
returning: Option[String])

object AlbumContestListResponse {
  import DateTimeCodecs._

  implicit val AlbumContestListResponseCodecJson: CodecJson[AlbumContestListResponse] = CodecJson.derive[AlbumContestListResponse]
  implicit val AlbumContestListResponseDecoder: EntityDecoder[AlbumContestListResponse] = jsonOf[AlbumContestListResponse]
  implicit val AlbumContestListResponseEncoder: EntityEncoder[AlbumContestListResponse] = jsonEncoderOf[AlbumContestListResponse]
}

File: client-clojure/src/sirqul_io_t_platform/specs/album_contest_list_response.clj
(ns sirqul-io-t-platform.specs.album-contest-list-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.name-string-value-response :refer :all]
            [sirqul-io-t-platform.specs.album-contest-response :refer :all]
            )
  (:import (java.io File)))


(def album-contest-list-response-data
  {
   (ds/opt :valid) boolean?
   (ds/opt :message) string?
   (ds/opt :version) float?
   (ds/opt :serializeNulls) boolean?
   (ds/opt :startTimeLog) int?
   (ds/opt :errorCode) string?
   (ds/opt :request) (s/coll-of name-string-value-response-spec)
   (ds/opt :items) (s/coll-of album-contest-response-spec)
   (ds/opt :hasMoreResults) boolean?
   (ds/opt :count) int?
   (ds/opt :countTotal) int?
   (ds/opt :returning) string?
   })

(def album-contest-list-response-spec
  (ds/spec
    {:name ::album-contest-list-response
     :spec album-contest-list-response-data}))
